packages/db-collections/tests/local-storage.test.ts

```diff
@@ -0,0 +1,654 @@
+      const existingTodos = {
+        "1": {
+          versionKey: `test-version-1`,
+          data: {
+            id: `1`,
+            title: `Existing Todo`,
+            completed: false,
+            createdAt: new Date(),
+          },
+        },
+      }
+
+      mockStorage.setItem(`todos`, JSON.stringify(existingTodos))
+
+      const collection = createCollection(
+        localStorageCollectionOptions<Todo>({
+          storageKey: `todos`,
+          storage: mockStorage,
+          storageEventApi: mockStorageEventApi,
+          getKey: (todo) => todo.id,
+        })
+      )
+
+      // Should load the existing data
+      expect(collection.size).toBe(1)
+      expect(collection.get(`1`)?.title).toBe(`Existing Todo`)
+    })
+
+    it(`should handle legacy array format in storage`, () => {
+      // Pre-populate storage with legacy array format
+      const existingTodos: Array<Todo> = [
+        {
+          id: `1`,
+          title: `Legacy Todo`,
+          completed: false,
+          createdAt: new Date(),
+        },
+      ]
+
+      mockStorage.setItem(`todos`, JSON.stringify(existingTodos))
+
+      const collection = createCollection(
+        localStorageCollectionOptions<Todo>({
+          storageKey: `todos`,
+          storage: mockStorage,
+          storageEventApi: mockStorageEventApi,
+          getKey: (todo) => todo.id,
+        })
+      )
+
+      // Should load and convert the legacy data
+      expect(collection.size).toBe(1)
+      expect(collection.get(`1`)?.title).toBe(`Legacy Todo`)
+    })
+
+    it(`should handle corrupted storage data gracefully`, () => {
+      // Set invalid JSON data
+      mockStorage.setItem(`todos`, `invalid json data`)
+
+      const collection = createCollection(
+        localStorageCollectionOptions<Todo>({
+          storageKey: `todos`,
+          storage: mockStorage,
+          storageEventApi: mockStorageEventApi,
+          getKey: (todo) => todo.id,
+        })
+      )
+
+      // Should initialize with empty collection
+      expect(collection.size).toBe(0)
+    })
+
+    it(`should handle empty storage gracefully`, () => {
+      const collection = createCollection(
+        localStorageCollectionOptions<Todo>({
+          storageKey: `todos`,
+          storage: mockStorage,
+          storageEventApi: mockStorageEventApi,
+          getKey: (todo) => todo.id,
+        })
+      )
+
+      // Should initialize with empty collection
+      expect(collection.size).toBe(0)
+    })
+  })
+
+  describe(`mutation handlers with storage operations`, () => {
+    it(`should perform insert operations and update storage`, async () => {
+      const collection = createCollection(
+        localStorageCollectionOptions<Todo>({
+          storageKey: `todos`,
+          storage: mockStorage,
+          storageEventApi: mockStorageEventApi,
+          getKey: (todo) => todo.id,
+          onInsert: () => Promise.resolve({ success: true }),
```

Comment:
Just re-factoring my app to use this and I was surprised to find this requiring an onInsert function to be configure in order to perform an insert.

My expectation was that it's a local storage backed in-memory collection and there's no need for a mutationFn. It's awesome that you can have a mutation function but would it not be better to default to a noop here rather than requiring onInsert, onUpdate and onDelete to be configured?

Or are you thinking that this would break consistency with other collection types and it's better to be explicit?